### ISSUE TITLE
Add a lint for dangerous parent selector use in mixins

### ISSUE THREAD

**Issue by nex3** [stylelint-scss/stylelint-scss#892]
It's relatively common for authors to write selectors like `.context &` as a way of logically scoping the current style rule to a given context. However, this isn't very reliable in mixins, because the mixin may be included within another nested style rule, at which point the `.context` may be misplaced.

It would be useful to have a lint for this case to help authors ensure they're not accidentally making assumptions in their mixins. It should specifically warn for `&` in style rules when:

* They're within a `@mixin`.
* They're nested within another style rule.
* They _aren't_ at the beginning of a given complex selector. (That is, `& .foo` is fine, as is `.foo, & .bar`.)
* ~~They _aren't_ within `@at-root`, since that can't be affected by parent style rules anyway.~~ (This isn't accurate, `&` is still affected by outer style rules.)

It would even be possible to suggest a fix for this by pulling the parent selector into the child and making the child style rule a sibling. For example:

```scss
@mixin foo {
  .parent {
    color: blue;

    .context & {
      color: red;
    }
  }
}
```

could be adjusted to:

```scss
@mixin foo {
  .parent {
    color: blue;
  }

  .context .parent {
    color: red;
  }
}
```

**Comment by kristerkari** [comment]
Why not, I guess.

How common is this? Sounds like a quite specific case.

**Comment by nex3** [comment]
It happens often enough in Google code (and causes enough problems) that we actively want a check for it.
